auth/src/main/java/software/amazon/awssdk/auth/signer/Aws4Signer.java

```diff
@@ -83,156 +66,90 @@
      * services that want to suppress this, they should use new
      * AWS4Signer(false).
      */
-    private boolean doubleUrlEncode;
-
-    private final SdkClock clock;
-
-    /**
-     * Construct a new AWS4 signer instance. By default, enable double
-     * url-encoding.
-     */
-    public Aws4Signer() {
-        this(true);
-    }
-
-    /**
-     * Construct a new AWS4 signer instance.
-     *
-     * @param doubleUrlEncoding Whether double url-encode the resource path when constructing
-     *                          the canonical request.
-     */
-    public Aws4Signer(boolean doubleUrlEncoding) {
-        this(doubleUrlEncoding, SdkClock.STANDARD);
-    }
-
-    @SdkTestInternalApi
-    public Aws4Signer(SdkClock clock) {
-        this(true, clock);
-    }
+    private final Boolean doubleUrlEncode;
```

Comment:
I still feel like this should be a per-request attribute, not something the person who is creating the signer needs to know. They (and we) should be able to reuse the signer between all services.

Reply:
I can move it to per-request attribute, but all services that use Aws4Signer set true for doubleUrlEncode. Only S3 which uses AwsS3V4Signer uses false. So I highly doubt it will change per request. But I am not hesitant to move it to per request attribute incase future new services want to disable double encoding in which case customers don't need to create new Signer instance for that service.

Reply:
+1 it won't change per-request, but neither would signing name, really. We can have a sensible default for doubleUrlEncode to cover for S3's weirdness.